library/CHART-TEMPLATE/tests/common_library_nodeselector_test.yaml

```diff
@@ -0,0 +1,40 @@
+suite: test nodeSelector helpers
+templates:
+  - templates/deployment.yaml
+release:
+  name: my-release
+  namespace: my-namespace
+set:
+  licenseKey: test-license-key
+  cluster: test-cluster
+tests:
+  - it: does not set nodeSelector if not specified
+    values:
+      - ./empty-values.yaml
+    set:
+    asserts:
+      - isNull:
+          path: spec.template.spec.nodeSelector
+  - it: does set nodeSelector if specified in values
+    values:
+      - ./empty-values.yaml
```

Comment:
you can set as well:
```
 - it: does not break if global=null
    set:
      global: null
    asserts:
      - equal:
[...]
```
This approach is better merely in the case we need ot test "weird" cases: for example "global.image" being null

Reply:
yes, i just changed it to that. 👍